### PR TITLE
[PDI-14705] Collator support for SortRows step

### PR DESCRIPTION
**Pull request by TheRealMadzor**
I've Implemented collator support for SortRows step.
I think i've edited all of the affected files and unit tests. I've also added unit tests for collation comparisons and of the extra entries for meta data inejction.

I've also updated the Entire names for SortRowsMetaInjection from IGNORE_CASE to CASE_SENSITIVE to be more in line with what you see in sortRows step properties.

### CODE REVIEW

File: core/src/org/pentaho/di/core/row/value/ValueMetaBase.java [hunk 1]
```diff
@@ -578,6 +588,70 @@ public void setCaseInsensitive( boolean caseInsensitive ) {
...
+  public String getCollatorStrength() {
+    return collatorStrength;
+  }
+
+  /**
+   * @param collatorStrength
+   *          the collatorStrength to set
+   */
+  @Override
+  public void setCollatorStrength( String collatorStrength ) {
```
Comment (mbatchelor) [inline core/src/org/pentaho/di/core/row/value/ValueMetaBase.java]: Should probably be an int which would eliminate the need to do the mapping this way. The only useful place for the string is in the UI classes.

File: core/src/org/pentaho/di/core/row/value/ValueMetaBase.java [hunk 2]
```diff
@@ -578,6 +588,70 @@ public void setCaseInsensitive( boolean caseInsensitive ) {
...
+        break;
+      case "TERTIARY":
+        this.collatorStrength = collatorStrength.toUpperCase();
+        this.collator.setStrength( 2 );
+        break;
+      case "IDENTICAL":
+        this.collatorStrength = collatorStrength.toUpperCase();
+        this.collator.setStrength( 3 );
+        break;
+      default:
```
Comment (mbatchelor) [inline core/src/org/pentaho/di/core/row/value/ValueMetaBase.java]: Disagree here - if they don't pass in a valid collator, it should throw an exception here.

File: engine/src/org/pentaho/di/trans/steps/sort/SortRowsMeta.java [hunk 3]
```diff
@@ -320,7 +341,8 @@ public void getFields( RowMetaInterface inputRowMeta, String name, RowMetaInterf
         ValueMetaInterface valueMeta = inputRowMeta.getValueMeta( idx );
         valueMeta.setSortedDescending( !ascending[i] );
         valueMeta.setCaseInsensitive( !caseSensitive[i] );
-
+        valueMeta.setCollatorDisabled( !collatorEnabled[i] );
```
Comment (mbatchelor) [inline engine/src/org/pentaho/di/trans/steps/sort/SortRowsMeta.java]: This kind of construct is awkward. You should stick with either collatorEnabled or collatorDisabled across the board. It's confusing when debugging to see xxxDisabled in one place, and then xxxEnabled in another.

File: ui/src/org/pentaho/di/ui/core/PropsUI.java [hunk 4]
```diff
@@ -200,7 +200,7 @@ protected synchronized void init() {
         }
       } catch ( ClassCastException cce ) {
         // Not all Lifecycle plugins implement GUIOption, keep calm and carry on
-        LogChannel.GENERAL.logDebug( "Plugin " + plugin.getIds()[0]
+        LogChannel.GENERAL.logBasic( "Plugin " + plugin.getIds()[0]
```
Comment (mbatchelor) [inline ui/src/org/pentaho/di/ui/core/PropsUI.java]: Not sure I agree with changing this to basic logging. Any particular reason you wanted to change this?

**Comment by mbatchelor** [comment]
@TheRealMadzor - Wow, thanks for the great contribution. I'm going to run it against our wingman server shortly, but I made some comments on the commit that I hope you can resolve. This looks like a really nice addition to SortRows!

**Comment by TheRealMadzor** [comment]
@mbatchelor  Thanks! I'll take a look at your comments as soon as i am able!

File: ui/src/org/pentaho/di/ui/core/PropsUI.java (hunk 4 above)
Comment (TheRealMadzor) [inline ui/src/org/pentaho/di/ui/core/PropsUI.java]: Not sure how that got in there, reverting.

File: engine/src/org/pentaho/di/trans/steps/sort/SortRowsMeta.java (hunk 3 above)
Comment (TheRealMadzor) [inline engine/src/org/pentaho/di/trans/steps/sort/SortRowsMeta.java]: Ok, i was following the existing convention, SortAcending vs Decending, CaseInsensitve vs Sensitive. Maybe i should change those as well?

**Comment by TheRealMadzor** [comment]
@mbatchelor I've made some changes and posted a comment ( a question ).

File: engine/src/org/pentaho/di/trans/steps/sort/SortRowsMeta.java (hunk 3 above)
Comment (mbatchelor) [inline engine/src/org/pentaho/di/trans/steps/sort/SortRowsMeta.java]: I understand, and I see that the pattern is as you've described. I hate it, but perhaps it's best to leave it as you have it.

**Comment by mbatchelor** [comment]
@TheRealMadzor - now that the 6.1 push is over, I'm able to re-look at this. Can you rebase so I can run this PR against Wingman.

**Comment by TheRealMadzor** [comment]
@mbatchelor I had some issues rebasing but i think i managed.

**Comment by eliofreitas** [comment]
Hello, This is an automated comment from our build system. Since this pull request originated from a user  not in the Pentaho organization, we will not proceed  with automatic validation of the pull request.  However, rest assured that it will be reviewed manually as soon  as we get the chance. Thanks for your contribution!

**Comment by matthewtckr** [comment]
Rebased and some minor changes in PR #2695

**Comment by mbatchelor** [comment]
@TheRealMadzor - Thanks so much for your contribution. Matt Tucker finished this one off, and I just merged it in!

**Comment by TheRealMadzor** [comment]
@mbatchelor  My pleasure!

**Pull request closed without merging**